src/sentry/releasehealth/base.py

```diff
@@ -23,13 +23,27 @@ class CurrentAndPreviousCrashFreeRate(TypedDict):
 CurrentAndPreviousCrashFreeRates = Mapping[ProjectId, CurrentAndPreviousCrashFreeRate]
 
 
+class _TimeBounds(TypedDict):
+    sessions_lower_bound: FormattedIsoTime
```

Comment:
Why doesn't this class look like this instead 
```
class _TimeBounds(TypedDict):
    sessions_lower_bound: Optional[FormattedIsoTime]
    sessions_upper_bound: Optional[FormattedIsoTime]
```

instead of having the class `_NoTimeBounds`

Reply:
Those are two different types. In my type, either all fields are non-None or all fields are None. In your type `lower_bound` can be `None` while `upper_bound` can be non-None.

Reply:
hmm I see although imo the usefulness of this is debatable but its up to you

Reply:
It's the correct type to have. In practice this means you can write code like:

```
if data['sessions_lower_bound'] is not None:
    # can assume upper bound is not None in here
```

instead of having to assert both fields for None/not-None all the time.

Reply:
Tbh I still wouldn't do the check like this because it might be confusing to read for whoever reads the code after me 

I would probably just check like 
```
if set(data.values()) == {None}:
    # do whatever
```